### DISCUSSION TITLE
ng update: the setup

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#345] [Announcements]
# ng update: the setup

Within 5 minutes you can automagically have your library upgraded when a user runs the ng update command.

https://timdeschryver.dev/blog/ng-update-the-setup

**Comment by neildeighan** [comment]
Hi Tim

I know this is an old post, and have used to try and get my solution working, and I have .. to .. a point ..

1) I have got the ng-add to work fine
2) I have got the ng update @myscope/mylib to work, i.e. just updates the package only

but .. I can only get migrations to work if I do this ..

ng update @myscope/mylib --migrate-only --from x.xx (and also adding --to x.xx)

so the above picks the correct schematics based on version in my migrations.json and runs them fine, so the setup must be ok

I thought by running

ng update @myscope/mylib@x.xx

would both update package and then run migrations automatically after, is this stil the case or has something changed ? Am i missing something ?

thanks for any insight you may have

**Reply by timdeschryver** [reply]
> ng update @myscope/[mylib@x.xx](mailto:mylib@x.xx)
> would both update package and then run migrations automatically after, is this stil the case or has something changed ? Am i missing something ?

This is correct, running `ng update` should update the library AND run the schematics between the as-is and to-be version. Can it be that the versions in the `migrations.json` doesn't match the upgraded version?

**Reply by neildeighan** [reply]
Hi Tim, just been going through the cli with the debugger, it appears that it gets the package info from the registry and not the package.json, my package is hosted on github, and when I do a fetch from https://npm.pkg.github.com/@myscope/mylib ... it brings back "most" of the standard "package.json" items, but NOT any extra i.e. ng-update ... so when the cli script tries to get  ['ng-update'] ... it is not there so just does the updates, and thinks there are no migrations .. however the migrateOnly reads the package.json, as I mentioned before the migrateOnly is working fine  .. so in summary

- I think my code is "ok", as it updates and migrates .. just not in one go .. but not because of my code.

- It looks like it might be a github thing maybe ? I cant see any docs around this yet, however i did note that @angular do not host on github packages they host on npmjs .. maybe for this reason?

- did try making my package public (has to be private) temporarily , but same info (less "ng-update" etc) comes back from fetch

- I havent tried publishing to npmjs to prove this yet, however npmjs is free for public, and not private, (doing on the cheap at moment ;) I will at some point .. but I got everything else in github at mo, i.e. packages/linked to repos, and mainly its free

- I can work around for now as I am using my "own cli" where I call custom schematics from so I can build my own "update logic" running  ng update .. and ng update --migrate-only to do the job

- question, when you was testing above post was you publishing to npmjs or github ?

**Reply by timdeschryver** [reply]
The packages were/are published to npm.
You can take a look at verdaccio to "run npm locally".

**Reply by neildeighan** [reply]
even if it works on npm or locally with verdaccio, however, ultimately it needs to be remote ( I could host my own verdaccio, but one more thing to look after) .. at least I know now why ... and I have an alternative workaround

**Reply by timdeschryver** [reply]
It seems like `ng add` supports providing a `--registry` (https://angular.dev/cli/add), but `ng update` does not. Sadly, I don't think there's a good solution for your case.

**Reply by neildeighan** [reply]
yeah my ng-add all good ... like i mentioned my ng update "works" I just have to run it twice, with the later using --migrate-only ... in my wrapper cli command I will sort out logic on which versions, as I will be in control of that, and my packages will not be for public consumption (so no complaints lol) .. was just so frustrating check over and over in the code to see what I have done wrong!
